r/R/table.R

```diff
@@ -185,6 +185,36 @@ Table <- R6Class("Table", inherit = ArrowObject,
   )
 )
 
+bad_attributes <- function(x) {
+  UseMethod("bad_attributes")
+}
+
+bad_attributes.default <- function(x) character()
+bad_attributes.data.frame <- function(x) c("class", "row.names", "names")
```

Comment:
Is this safe to do? `sf`, for example, creates a subclass of `data.frame`: https://github.com/r-spatial/sf/blob/master/R/sf.R#L287

But since this works by inheritance, we'll drop that class attribute. 

It's (arguably) uglier, but maybe it's better to have a function with a bunch of if-else rather than use S3 methods, and then we can distinguish `identical(class(x), "data.frame")` from `inherits(x, "data.frame")`.

Reply:
We might actually want to keep the class for data frames, so that we would faithfully roundtrip data frames, instead of enforcing a tibble: 

``` r
library(arrow, warn.conflicts = TRUE)
#> 
#> Attaching package: 'arrow'
#> The following object is masked from 'package:utils':
#> 
#>     timestamp

as.data.frame(record_batch(df = data.frame(x = 1)))$df
#> # A tibble: 1 x 1
#>       x
#>   <dbl>
#> 1     1
```

<sup>Created on 2020-09-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

Reply:
And if we do that, we get to reconstruct grouped data frames as well: 

``` r
library(arrow, warn.conflicts = FALSE)
library(dplyr, warn.conflicts = FALSE)

df <- data.frame(x = 1:10, g = rep(1:2, each = 5)) %>% group_by(g)
as.data.frame(Table$create(df))
#> # A tibble: 10 x 2
#> # Groups:   g [2]
#>        x     g
#>    <int> <int>
#>  1     1     1
#>  2     2     1
#>  3     3     1
#>  4     4     1
#>  5     5     1
#>  6     6     2
#>  7     7     2
#>  8     8     2
#>  9     9     2
#> 10    10     2
```

<sup>Created on 2020-09-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>